prog/weaver/main.go

```diff
@@ -452,18 +452,21 @@ func main() {
 		go expose(allocator, defaultSubnet, bridgeConfig.WeaveBridgeName, bridgeConfig.AWSVPC)
 	}
 
-	// Start Weave Proxy:
-	proxy, err := weaveproxy.NewProxy(*proxyConfig)
-	if err != nil {
-		Log.Fatalf("Could not start proxy: %s", err)
+	if proxyConfig.Disabled {
+		signals.SignalHandlerLoop(common.Log, router)
+	} else {
+		// Start Weave Proxy:
+		proxy, err := weaveproxy.NewProxy(*proxyConfig)
+		if err != nil {
+			Log.Fatalf("Could not start proxy: %s", err)
+		}
+		defer proxy.Stop()
+		listeners := proxy.Listen()
+		proxy.AttachExistingContainers()
+		go proxy.Serve(listeners)
+		go proxy.ListenAndServeStatus("/home/weave/status.sock")
+		signals.SignalHandlerLoop(common.Log, proxy, router)
```

Comment:
Purely because it isn't there in [`master`](https://github.com/weaveworks/weave/blob/master/prog/weaver/main.go#L448) at the moment and I didn't want to diverge too much from it given I'm still discovering the codebase, and I had enough troubles to get it to a stable state already.
I can add it though, now that all smoke tests are passing.